shell/platform/linux/fl_view.cc

```diff
@@ -19,13 +22,194 @@ struct _FlView {
   FlRendererX11* renderer;
   FlEngine* engine;
   int64_t button_state;
+
+  // Flutter system channels.
+  FlKeyEventChannel* key_event_channel;
 };
 
 enum { PROP_FLUTTER_PROJECT = 1, PROP_LAST };
 
 G_DEFINE_TYPE(FlView, fl_view, GTK_TYPE_WIDGET)
 
-// Convert a GDK button event into a Flutter event and send to the engine
+// Converts a Gdk key code to its GLFW equivalent.
+// TODO(robert-ancell) Create a "gtk" toolkit in Flutter so we don't have to
```

Comment:
You can reference https://github.com/flutter/flutter/issues/57603 here, which I just turned into an issue.